.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,93 @@
+          "$GITHUB_REF_NAME" \
+          "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY" \
+          "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" > version.json
+      - name: Output version.json
+        run: cat version.json
+      - name: Build Docker images
+        run: make build
+      - name: Verify requirements.txt contains correct dependencies
+        run: |
+          docker compose run --rm --no-deps ci shell ./bin/run_verify_reqs.sh
+      - name: Run lint check
+        run: |
+          make my.env
+          docker compose run --rm --no-deps ci shell ./bin/run_lint.sh
+      - name: Run tests
+        run: |
+          make my.env
+          docker compose up --detach --no-color \
+              localstack \
+              statsd \
+              fakesentry
+          docker compose run --rm ci shell ./bin/run_tests.sh
+      - name: Run systemtest
+        run: |
+          docker compose run --rm ci-web shell ./bin/run_setup.sh
+          docker compose up --detach --wait --wait-timeout=10 ci-web
+          docker compose run --rm ci-web shell bash -c 'cd systemtest && NGINX_TESTS=0 POST_CHECK=1 HOST=http://ci-web:8000 pytest -vv'
+      - name: Run systemtest with pubsub and gcs
+        run: |
+          echo 'CRASHMOVER_CRASHPUBLISH_CLASS=antenna.ext.pubsub.crashpublish.PubSubCrashPublish' >> my.env
+          echo 'CRASHMOVER_CRASHSTORAGE_CLASS=antenna.ext.gcs.crashstorage.GcsCrashStorage' >> my.env
+          docker compose run --rm ci-web shell ./bin/run_setup.sh
+          docker compose up --detach --wait --wait-timeout=10 ci-web
+          # Use -m "not aws" to select gcp and unmarked tests
+          docker compose run --rm ci-web shell bash -c 'cd systemtest && NGINX_TESTS=0 POST_CHECK=1 HOST=http://ci-web:8000 pytest -vv -m "not aws"'
+          # remove config on last two lines
+          sed '$d' -i my.env
+          sed '$d' -i my.env
+
+      - name: Set Docker image tag to "latest" for updates of the main branch
+        if: github.ref == 'refs/heads/main'
+        run: |
+          echo IMAGE_TAG=latest >> "$GITHUB_ENV"
+          # Updates to the main branch are deployed to stage.
+          echo DEPLOYMENT_ENV=stage >> "$GITHUB_ENV"
+      - name: Set Docker image tag to the git tag for tagged builds
+        if: startsWith(github.ref, 'refs/tags/')
+        run: |
+          echo IMAGE_TAG="$GITHUB_REF_NAME" >> "$GITHUB_ENV"
+          # Version tags are deployed to prod.
+          echo DEPLOYMENT_ENV=prod >> "$GITHUB_ENV"
+      - name: Push the Docker image to GAR
+        if: env.IMAGE_TAG != ''
+        uses: mozilla-it/deploy-actions/docker-push@v3.9.0
+        with:
+          local_image: local/antenna_deploy_base:latest
+          image_repo_path: ${{ secrets.DOCKER_IMAGE_PATH }}
+          image_tag: ${{ env.IMAGE_TAG }}
+          workload_identity_pool_project_number: ${{ secrets.WORKLOAD_IDENTITY_POOL_PROJECT_NUMBER }}
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
```

Comment:
I configured these secrets, but i don't know if i got them right, i will need to confirm with @smarnach when he is back

Reply:
figured it out